mindmap/tests/test_mindmap.py

```diff
@@ -131,6 +221,253 @@ def test_studio_view(self):
+            "is_static": True,
+            "points": 100,
+            "weight": 1,
+        }
+        self.request = Mock(
+            body=json.dumps(self.data).encode("utf-8"),
+            method="POST",
+            status_code_success=200,
+        )
+        self.student_id = "test-student-id"
+        self.grade = 100
+        self.submission_id = "test-submission-id"
+
+    def test_studio_submit(self):
+        """
+        Check studio submit handler.
+
+        Expected result:
+            - The studio view is rendered with the appropriate values.
+        """
+        self.xblock.studio_submit(self.request)
+
+        self.assertEqual(self.data["display_name"], self.xblock.display_name)
+        self.assertEqual(self.data["mind_map"], self.xblock.mindmap_body)
+        self.assertEqual(self.data["is_static"], self.xblock.is_static)
+        self.assertEqual(self.data["points"], self.xblock.points)
+        self.assertEqual(self.data["weight"], self.xblock.weight)
+
+    @ddt.data(
+        {"points": "/-100", "weight": 1},
+        {"points": 100, "weight": ".-0.5"},
+        {"points": "/100", "weight": "/0.5"},
+    )
+    def test_studio_submit_badly_formatted(self, bad_formatted_data: dict):
+        """
+        Check studio submit handler with badly formatted data.
+
+        Expected result:
+            - The studio view raises an error.
+        """
+        data = {
+            "display_name": "Test Mind Map",
+            "mindmap_body": self.mind_map,
+            "is_static": True,
+        }
+        data.update(bad_formatted_data)
+        self.request.body = json.dumps(data).encode("utf-8")
+
+        response = self.xblock.studio_submit(self.request)
+
+        self.assertEqual(400, response.status_code)
+
+    def test_save_assignment(self):
+        """
+        Check save assignment JSON handler.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The mind map is saved.
+        """
+        response = self.xblock.save_assignment(self.request)
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(self.data["mind_map"], self.xblock.mindmap_student_body)
+
+    @patch("submissions.api.create_submission")
+    def test_submit_assignment(self, create_submission_mock: Mock):
+        """
+        Check submit assignment handler.
+
+        Expected result:
+            - The student view is rendered with the appropriate values.
+        """
+        expected_student_item_dict = {
+            "item_id": self.xblock.block_id,
+            "item_type": "mindmap",
+            "student_id": self.xblock.get_current_user().opt_attrs.get(),
+            "course_id": self.xblock.block_course_id,
+        }
+        expected_answer = {
+            "mindmap_student_body": json.dumps(self.data["mind_map"]),
+        }
+
+        response = self.xblock.submit_assignment(self.request)
+
+        self.assertEqual(200, response.status_code)
```

Comment:
Note to author: change this to constant from django / python / requests

Reply:
Maybe you can use `http` module:

```python
from http import HTTPStatus

self.assertEqual(HTTPStatus.OK, response.status_code)
```